src/Uno.UWP/Media/SpeechRecognition/SpeechRecognizer.wasm.cs

```diff
@@ -0,0 +1,123 @@
+{
+	public partial class SpeechRecognizer
+	{
+		private const string JsType = "Windows.Media.SpeechRecognizer";
+
+		private readonly static ConcurrentDictionary<string, SpeechRecognizer> _instances =
+			new ConcurrentDictionary<string, SpeechRecognizer>();
+
+		private readonly Guid _instanceId = Guid.NewGuid();
+
+		private TaskCompletionSource<SpeechRecognitionResult> _currentCompletionSource;
+
+		[Preserve]
+		public static int DispatchStatus(string instanceId, string state)
+		{
+			if (_instances.TryGetValue(instanceId, out var speechRecognizer))
+			{
+				if (Enum.TryParse<SpeechRecognizerState>(state, true, out var stateEnum))
+				{
+					speechRecognizer.OnStateChanged(stateEnum);
+				}
+			}
+			return 0;
+		}
+
+		[Preserve]
+		public static int DispatchError(string instanceId, string error)
+		{
+			if (_instances.TryGetValue(instanceId, out var speechRecognizer))
+			{
+				if (speechRecognizer._currentCompletionSource != null)
+				{
+					speechRecognizer._currentCompletionSource.SetException(
+						new InvalidOperationException($"Speech recognition failed with '{error}'"));
+				}
+				else
+				{
+					if (typeof(SpeechRecognizer).Log().IsEnabled(Microsoft.Extensions.Logging.LogLevel.Error))
+					{
+						typeof(SpeechRecognizer).Log().LogError($"Speech recognition failed with '{error}'");
+					}
+				}
+			}
+			return 0;
+		}
+
+		[Preserve]
+		public static int DispatchHypothesis(string instanceId, string hypothesis)
+		{
+			if (_instances.TryGetValue(instanceId, out var speechRecognizer))
+			{
+				speechRecognizer.OnHypothesisGenerated(hypothesis);
+			}
+			return 0;
+		}
+
+		[Preserve]
+		public static int DispatchResult(string instanceId, string result, double confidence)
+		{
+			if (_instances.TryGetValue(instanceId, out var speechRecognizer))
+			{
+				speechRecognizer.OnStateChanged(SpeechRecognizerState.Idle);
+				var recognitionResult = new SpeechRecognitionResult()
+				{
+					Text = result,
+					RawConfidence = confidence
+				};
+				speechRecognizer?._currentCompletionSource.SetResult(recognitionResult);
+			}
+			return 0;
+		}
+
+		public IAsyncOperation<SpeechRecognitionResult> RecognizeAsync() =>
+			RecognizeTaskAsync().AsAsyncOperation();
+
+		private async Task<SpeechRecognitionResult> RecognizeTaskAsync()
+		{
+			var existingTask = _currentCompletionSource?.Task;
+			if (existingTask != null)
+			{
+				return await existingTask;
+			}
+
+			_currentCompletionSource = new TaskCompletionSource<SpeechRecognitionResult>();
+
+			var command = $"{JsType}.recognize('{_instanceId}')";
+			var recognizeResult = WebAssemblyRuntime.InvokeJS(command);
+			if (!bool.TryParse(recognizeResult, out var canRecognize) || !canRecognize)
+			{
+				throw new InvalidOperationException(
+					"Speech recognizer is not available on this device.");
+			}
+
+			var result = await _currentCompletionSource.Task;
+			_currentCompletionSource = null;
+			return result;
+		}
+
+		public void Dispose()
```

Comment:
Codacy found an issue: [Either implement 'IDisposable.Dispose', or totally rename this method to prevent confusion.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5934933)